app/controllers/pipeline_outputs_controller.rb

```diff
@@ -29,6 +29,8 @@ def create
 
     respond_to do |format|
       if @pipeline_output.save
+        @pipeline_output.name = ['ID#', @pipeline_output.id, ' (', @pipeline_output.sample.name, ')'].join('')
```

Comment:
why not do this before the first save?

Reply:
note that, as written, if the second save fails we'll not handle the error gracefully

Reply:
Can't access pipeline_output.id until it's been saved!

Reply:
Oh i understand now.

There's no reason to have a `name` column in the db. Just create a method called `#name` in PipelineOutput like so:

```
def name
  return ['ID#', self.id, ' (', self.sample.name, ')'].join('')
end
```